### DISCUSSION TITLE
Microsoft Teams integration for Cypress Cloud

### DISCUSSION THREAD

**Discussion by pstakoun** [cypress-io/cypress#26795] [General]
Today, we’re excited to release our new integration with Microsoft Teams. For more information about this integration, [check out our Docs](http://on.cypress.io/teams-integration).

We’d love to hear your feedback on this new integration:

- How is your team using this integration?
- Did you find the integration process to be intuitive?
- What features might make this integration more valuable?

**Comment by ElGibbo** [comment]
Have just started to use this feature today, love the amount of detail on Teams.
It would be good if the setup options had the ability to filter on Project, and also to filter by branch.

**Reply by mattvollmer** [reply]
Hi @ElGibbo --

To clarify, do you mean it would be nice to trigger Teams notifications for specific Projects (and branches)?

If yes, you can currently configure Teams per Project by navigating to your Project, then Project Settings:
![image](https://github.com/cypress-io/cypress/assets/95866673/0cfba855-2a92-4378-bf61-c9ed787a4192)

To your point, branch filtering is not currently supported. Thanks for surfacing that one.

**Reply by ElGibbo** [reply]
@mattvCypress Brilliant, hadn't seen this. Thank you.

**Comment by miguelpujadas** [comment]
Hi! Thanks for the integration!
Any idea if it will be possible in a future to filter by tags?
We have different suites and we would like to send messages to different channels depending on the tags.

**Reply by mattvollmer** [reply]
This is a feature @pstakoun can chime in on, but I know this is on the radar.

**Reply by pstakoun** [reply]
@miguelpujadas thanks for the feedback. This is indeed something that is planned.

**Reply by jwwitherspoon** [reply]
@pstakoun are there any issues open for contribution surrounding this feature?

**Reply by pstakoun** [reply]
@jwwitherspoon this feature is part of [Cypress Cloud](https://www.cypress.io/cloud) and thus is not part of the open source repo, but all feedback is considered as we prioritize upcoming improvements.

**Comment by marijaruiz** [comment]
Any plan on enabling more settings for this integration? For example controlling what kind of information is reported in the channel like tags, or editing the contents of the message?

**Comment by Cleudi** [comment]
Quite some time has passed by, are there any news regarding tag filtering, @pstakoun ?
I can't imagine that this is a complicated piece of work, since it's already implemented for the slack integration for years...

When working with different staging systems, having tag filters is absolutely necessary.
